### DISCUSSION TITLE
Using ruleset_converter Outputs Many Errors

### DISCUSSION THREAD

**Discussion by faxotherapy** [bromite/bromite#1683] [Q&A]
Hi, I wanted to experiment the [ruleset_converter](https://www.bromite.org/custom-filters) tool:
```
$ ./ruleset_converter --input_format=filter-list --output_format=unindexed-ruleset --input_files=badware.txt,filters.txt,privacy.txt,resource-abuse.txt,unbreak.txt,easylist.txt,easyprivacy.txt,Peter_Lowe.txt --output_file=filters.dat
```

Althtough the generated DAT files works fine with Bromite, there are many output errors during tool execution, such as:
```
[1227/011204.665847:ERROR:rule_stream.cc(88)] (error:2) Unknown URL rule option:
$csp=worker-src 'none',domain=torlock.com|alltube.pl|alltube.tv|centrum-dramy.pl|cmovies.ac|coinfaucet.eu|crictime.com|crictime.is|deltabit.co|doodcdn.com|estream.to|extreme-down.tv|flashx.co|flashx.to|flashx.tv|gomo.to|hdvid.fun|hdvid.tv|hitomi.la|kinox.to|lewd.ninja|assia1.tv|assia2.tv|nflbite.com|pirateproxy.live|plytv.me|potomy.ru|powvideo.cc|powvideo.net|putlocker.to|reactor.cc|rojadirecta.watch|sickrage.ca|streamtape.com|thehiddenbay.com|thepiratebay.org|thepiratebay10.org|tpb.party|uptomega.me|ustream.to|vidoza.co|vidoza.net|vidtodo.pro|wallpoper.com|wearesaudis.net|yazilir.com
 ^

[1227/011204.666160:ERROR:rule_stream.cc(88)] (error:20) Unknown URL rule option:
||convertcase.net^$csp=script-src 'self' '*' blob: data:
```

Should I ignore those messages?

Thanks for your help.

**Comment by Helium314** [comment]
I just encountered the same issue.
This happens because the ruleset converter doesn't understand all rules (in this case the `csp` rule). Nothing you can do in this case.

However, Bromite is using the filters from https://github.com/bromite/filters/blob/master/lists.txt, but the uBlock lists contain aliases such as `3p` being used instead of `third-party`.
@csagan5 are the aliases converted before running `ruleset_converter`? I didn't find any information on this.

If they are considered, could the necessary conversion be made available somewhere (e.g. on the wiki or on https://www.bromite.org/custom-filters)?
